benchmark/v6.2-performance-benchmarking-with-tpcc.md

```diff
@@ -0,0 +1,123 @@
+### 软件版本
+
+| 服务类型 | 软件版本       |
+| :------- | :------------- |
+| PD       | v6.1.0、v6.2.0 |
+| TiDB     | v6.1.0、v6.2.0 |
+| TiKV     | v6.1.0、v6.2.0 |
+| TiUP     | 1.9.3          |
+| HAProxy  | 2.5.0          |
+
+### 配置参数
+
+两个版本使用同样的配置。
+
+#### TiDB 参数配置
+
+{{< copyable "" >}}
+
+```yaml
+log.level: "error"
+prepared-plan-cache.enabled: true
+tikv-client.max-batch-wait-time: 2000000
+```
+
+#### TiKV 参数配置
+
+{{< copyable "" >}}
+
+```yaml
+raftstore.apply-max-batch-size: 2048
+raftstore.apply-pool-size: 3
+raftstore.store-max-batch-size: 2048
+raftstore.store-pool-size: 2
+readpool.storage.normal-concurrency: 10
+server.grpc-concurrency: 6
+```
+
+#### TiDB 全局变量配置
+
+{{< copyable "sql" >}}
+
+```sql
+set global tidb_hashagg_final_concurrency=1;
+set global tidb_hashagg_partial_concurrency=1;
+set global tidb_enable_async_commit = 1;
+set global tidb_enable_1pc = 1;
+set global tidb_guarantee_linearizability = 0;
+set global tidb_enable_clustered_index = 1;
+set global tidb_prepared_plan_cache_size=1000;
+```
+
+#### HAProxy 配置 - haproxy.cfg 文件
+
+更多有关 HAProxy 在 TiDB 上的使用，可参阅 [HAProxy 在 TiDB 中的最佳实践](/best-practices/haproxy-best-practices.md)。
+
+{{< copyable "" >}}
+
+```yaml
+global                                     # 全局配置。
+   pidfile     /var/run/haproxy.pid        # 将 HAProxy 进程的 PID 写入 pidfile。
+   maxconn     4000                        # 每个 HAProxy 进程所接受的最大并发连接数。
+   user        haproxy                     # 同 UID 参数。
+   group       haproxy                     # 同 GID 参数，建议使用专用用户组。
+   nbproc      64                          # 在后台运行时创建的进程数。在启动多个进程转发请求时，确保该值足够大，保证 HAProxy 不会成为瓶颈。
+   daemon                                  # 让 HAProxy 以守护进程的方式工作于后台，等同于命令行参数“-D”的功能。当然，也可以在命令行中用“-db”参数将其禁用。
+
+defaults                                   # 默认配置。
+   log global                              # 日志继承全局配置段的设置。
+   retries 2                               # 向上游服务器尝试连接的最大次数，超过此值便认为后端服务器不可用。
+   timeout connect  2s                     # HAProxy 与后端服务器连接超时时间。如果在同一个局域网内，可设置成较短的时间。
+   timeout client 30000s                   # 客户端与 HAProxy 连接后，数据传输完毕，即非活动连接的超时时间。
+   timeout server 30000s                   # 服务器端非活动连接的超时时间。
+
+listen tidb-cluster                        # 配置 database 负载均衡。
+   bind 0.0.0.0:3390                       # 浮动 IP 和 监听端口。
+   mode tcp                                # HAProxy 要使用第 4 层的传输层。
+   balance leastconn                      # 连接数最少的服务器优先接收连接。`leastconn` 建议用于长会话服务，例如 LDAP、SQL、TSE 等，而不是短会话协议，如 HTTP。该算法是动态的，对于启动慢的服务器，服务器权重会在运行中作调整。
+   server tidb-1 10.9.18.229:4000 check inter 2000 rise 2 fall 3       # 检测 4000 端口，检测频率为每 2000 毫秒一次。如果 2 次检测为成功，则认为服务器可用；如果 3 次检测为失败，则认为服务器不可用。
+   server tidb-2 10.9.39.208:4000 check inter 2000 rise 2 fall 3
+   server tidb-3 10.9.64.166:4000 check inter 2000 rise 2 fall 3
+```
+
+## 测试方案
+
+1. 通过 TiUP 部署 TiDB v6.2.0 和 v6.1.0。
+2. 创建数据库 tpcc：`create database tpcc;`。
+3. 通过 tiup bench 导入 TPC-C 5000 Warehouse 数据：`tiup bench tpcc prepare --warehouses 5000 --db tpcc -H 127.0.0.1 -p 4000`。
```

Comment:
@Yui-Song According to https://github.com/pingcap/docs/pull/8903, `p` should be upper-case. Is that correct?
```suggestion
3. 通过 tiup bench 导入 TPC-C 5000 Warehouse 数据：`tiup bench tpcc prepare --warehouses 5000 --db tpcc -H 127.0.0.1 -P 4000`。
```